c7n/schema.py

```diff
@@ -182,6 +183,8 @@ def generate(resource_types=()):
                     'type': 'string',
                     'pattern': "^[A-z][A-z0-9]*(-[A-z0-9]+)*$"},
                 'region': {'type': 'string'},
+                'start': {'format': 'date'},
```

Comment:
Yeah, it prevents passing in nonsense like `asdf` into the field. There's also support for a bunch of other format types that you can use with `jsonschema.FormatCheckers().checkers`

Getting some issues with running `custodian validate` though, for some reason it's not using the schema defined in `c7n/schema.py`.

```
2018-03-30 15:07:49,250: custodian.commands:ERROR Configuration invalid: test.yml
2018-03-30 15:07:49,252: custodian.commands:ERROR Additional properties are not allowed ('start' was unexpected)

Failed validating u'additionalProperties' in schema[31][u'allOf'][0]:
    {u'additionalProperties': False,
     u'properties': {u'actions': {u'type': u'array'},
                     u'comment': {u'type': u'string'},
                     u'comments': {u'type': u'string'},
                     u'description': {u'type': u'string'},
                     u'filters': {u'type': u'array'},
                     u'max-resources': {u'type': u'integer'},
                     u'mode': {u'$ref': u'#/definitions/policy-mode'},
                     u'name': {u'pattern': u'^[A-z][A-z0-9]*(-[A-z0-9]+)*$',
                               u'type': u'string'},
                     u'query': {u'items': {u'maxProperties': 1,
                                           u'minProperties': 1,
                                           u'type': u'object'},
                                u'type': u'array'},
                     u'region': {u'type': u'string'},
                     u'resource': {u'type': u'string'},
                     u'source': {u'enum': [u'describe', u'config']},
                     u'tags': {u'items': {u'type': u'string'},
                               u'type': u'array'}},
     u'required': [u'name', u'resource'],
     u'type': u'object'}

On instance:
    {'name': 'expire-03-29',
     'resource': 'ec2',
     'start': datetime.date(2020, 3, 30)}
2018-03-30 15:07:49,252: custodian.commands:ERROR expire-03-29
```

Reply:
we automatically pickup all the format checkers from draft4 of jsonschema.  the issue with passing checkers explicitly is that it can affect which standard (if any) we're actually complying with, also afaics jsonschema validation should never be returning transformed values.

Reply:
its definitely transforming the values.